### PR TITLE
feat: add bitbucket

### PR DESCRIPTION
**Pull request by beatreichenbach**
This pull request adds support for the Bitbucket remote vcs. Bitbucket doesn't support releases like Github but I added a skeleton class anyway so that users can use `semantic-release version` without having to use the `--no-vcs-release` flag. It still pushes a tag to the remote vcs which creates archives so it's nice to be able to use semantic-release without necessarily having actual releases in the remote vcs.

### CODE REVIEW

**Comment by codejedi365** [comment]
This is exciting! Thanks for the work! Recommend adding BitBucket as one of the Hvcs clients tested in:
`tests/unit/semantic_release/changelog.py::test_changelog_context()`
to ensure rendering can occur with the default changelogs.

**Comment by beatreichenbach** [comment]
> Recommend adding BitBucket as one of the Hvcs clients tested in: `tests/unit/semantic_release/changelog.py::test_changelog_context()` to ensure rendering can occur with the default changelogs.

Thank you!
I added the test and it all seems to work. (I also have been using it in "production" and the changelog has been created fine.)

**Comment by beatreichenbach** [comment]
That's good to know, thanks. I never used different strategies. I'll keep an eye out for that and will play around with different setups to see if i can get anything to break.

**Review by codejedi365 — COMMENTED** [review]
Good work on this. A few comments and I'd also recommend adding a few comments to `docs/index.rst` related to the environment variables required for BitBucket.

Otherwise I pass it over to @bernardcooke53 as one of the maintainers.

File: tests/unit/semantic_release/hvcs/test_bitbucket.py
Comment (beatreichenbach) [reply tests/unit/semantic_release/hvcs/test_bitbucket.py:71]: Thank you, I copied everything from the existing tests 🙃

File: semantic_release/hvcs/bitbucket.py
Comment (beatreichenbach) [reply semantic_release/hvcs/bitbucket.py:87]: Thank you so much for your help with all of this. This is my first real contribution to an open source project, so I really appreciate your detailed explanations, very kind!
I added a section to the `docs/index.rst` to document the only extra environment variable for bitbucket: `BITBUCKET_USER`. Maybe it's not the best solution but it's something until we come up with anything better.

File: semantic_release/hvcs/bitbucket.py
Comment (beatreichenbach) [reply semantic_release/hvcs/bitbucket.py]: You're absolutely right, I copied it from GitHub and removed it, as it makes no sense to use undocumented, unofficial environment variables.

File: semantic_release/hvcs/bitbucket.py
Comment (beatreichenbach) [reply semantic_release/hvcs/bitbucket.py]: You're absolutely right, I copied it from GitHub and removed it, as it makes no sense to use undocumented, unofficial environment variables.

File: semantic_release/hvcs/bitbucket.py
Comment (codejedi365) [reply semantic_release/hvcs/bitbucket.py:87]: Congratulations! It's a great feeling and I'm glad you made the endeavor. Glad I could help, and the effort you put forth in adding testing and docs is much more than most do, so keep it up!

Comment (bernardcooke53) [reply semantic_release/hvcs/bitbucket.py:87]: I have to echo the sentiment from @codejedi365 - as far as contributions go this one is really excellent @beatreichenbach, particularly for a first-time contribution too!

We will eventually have to do something about the growing number of CI systems and Git remotes. Eventually it would be really nice to rekindle the talk of plugin architectures...

File: docs/configuration.rst
Comment (bernardcooke53) [inline docs/configuration.rst:1]: I think the table of default environment variables in `token` needs to be updated too

**Review by bernardcooke53 — CHANGES_REQUESTED** [review]
This is a really great piece of work! I'd ask that you take a look at the linter/test failures before merging, but thank you for the effort to make the contribution! 🎉

File: docs/configuration.rst
Comment (beatreichenbach) [reply docs/configuration.rst:1]: Thanks!
I forgot to run the tests after I removed the environment variables. I removed the environment variable tests and committed the new change. Apologies for missing that!

> I think the table of default environment variables in token needs to be updated too

Can you please point me to that? I'm not sure I understand what you mean.

Comment (codejedi365) [reply docs/configuration.rst:1]: It's a documentation change on the configuration page (it's now a table from the merge #786 today). We attempt to guess the `remote.token` value based on the VCS chosen.

Comment (beatreichenbach) [reply docs/configuration.rst:1]: Thank you again for all the help! This is fantastic.

I rebased my changes (i definitely should have made a branch for this, working in master was not smart.)
And I added the bitbucket token to the table in the docs.

There are a couple lint errors but they should be okay to ignore. Let me know if you want 0 errors and I'll fix those.

**Comment by beatreichenbach** [comment]
Any updates on this?

**Comment by beatreichenbach** [comment]
> LGTM - going to hang fire on merging until the package publishing to PyPI is fixed. Thanks @beatreichenbach!

Thank you so much! No rush on merging.

**Comment by codejedi365** [comment]
@beatreichenbach, the pypi publishing is resolved. Recommend rebasing when you have a chance and updating the lint error that occured.

**Comment by beatreichenbach** [comment]
> @beatreichenbach, the pypi publishing is resolved. Recommend rebasing when you have a chance and updating the lint error that occured.

Fixed the ruff linter and rebased onto latest commit.

**Comment by codejedi365** [comment]
@relekang

**Pull request merged**
